vmm/src/lib.rs

```diff
@@ -265,6 +265,7 @@ impl Vmm {
         if let Some(ref mut vm) = self.vm {
             let config = vm.get_config();
             vm.shutdown()?;
+            self.vm = None;
```

Comment:
That's a correct fix, thanks.
However, in order to use the already factorized code and since `vm_shutdown` already call `take` on the vm reference, I'd prefer if it would be:

```Rust
diff --git a/vmm/src/lib.rs b/vmm/src/lib.rs
index 75f3653..4da0458 100644
--- a/vmm/src/lib.rs
+++ b/vmm/src/lib.rs
@@ -256,7 +256,7 @@ impl Vmm {
         #[cfg(not(feature = "acpi"))]
         {
             if let Some(ref mut vm) = self.vm {
-                vm.shutdown()?;
+                self.vm_shutdown()?;
                 return Ok(());
             }
         }
@@ -264,7 +264,7 @@ impl Vmm {
         // First we stop the current VM and create a new one.
         if let Some(ref mut vm) = self.vm {
             let config = vm.get_config();
-            vm.shutdown()?;
+            self.vm_shutdown()?;
 
             let exit_evt = self.exit_evt.try_clone().map_err(VmError::EventFdClone)?;
             let reset_evt = self.reset_evt.try_clone().map_err(VmError::EventFdClone)?;
```

Reply:
@sameo Update is done!